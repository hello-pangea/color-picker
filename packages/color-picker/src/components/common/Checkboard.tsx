import React, { isValidElement } from "react";
import * as checkboard from "../../helpers/checkboard";

export type CheckboardProps = {
  size?: number;
  white?: string;
  grey?: string;
  renderers?: any;
  borderRadius?: string | number;
  boxShadow?: string;
  children?: React.ReactNode;
};

export default function Checkboard({
  white = "transparent",
  grey = "rgba(0,0,0,.08)",
  size = 8,
  renderers = {},
  borderRadius,
  boxShadow,
  children,
}: CheckboardProps) {
  const styles: Record<string, React.CSSProperties> = {
    grid: {
      borderRadius,
      boxShadow,
      position: "absolute",
      inset: "0px",
      background: `url(${checkboard.get(
        white,
        grey,
        size,
        renderers.canvas
      )}) center left`,
    },
  };

  return isValidElement(children) ? (
    React.cloneElement(children, {
      ...children.props,
      style: { ...children.props.style, ...styles.grid },
    })
  ) : (
    <div style={styles.grid} />
  );
}
