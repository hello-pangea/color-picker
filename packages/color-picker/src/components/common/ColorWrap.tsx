import React from "react";
import ColorProvider, {
  useColor,
  withColorProvider,
} from "../../context/useColor";
import { Color, ColorResult } from "../../types/colors";

export type InjectedColorProps = {
  onSwatchHover?: (color: ColorResult, event: React.MouseEvent) => void;
  onChangeComplete?: (color: ColorResult) => void;
  color?: Color;
  defaultColor?: Color;
};

export const ColorWrap = (Picker: any) => {
  function ColorPicker({
    onSwatchHover,
    onChangeComplete,
    color,
    defaultColor,
  }: InjectedColorProps) {
    const { colors, changeColor } = useColor();

    return (
      <ColorProvider
        color={color}
        defaultColor={defaultColor}
        onChangeComplete={onChangeComplete}
        onSwatchHover={onSwatchHover}
      >
        <Picker {...colors} onChange={changeColor} />
      </ColorProvider>
    );
  }

  return withColorProvider(ColorPicker);
};

export default ColorWrap;
