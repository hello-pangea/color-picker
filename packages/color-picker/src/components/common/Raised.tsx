import merge from "lodash/merge";
import React from "react";

export type RaisedProps = {
  background?: string;
  zDepth?: 0 | 1 | 2 | 3 | 4 | 5;
  radius?: number;
  styles?: Record<string, React.CSSProperties>;
  children?: React.ReactNode;
};

export default function Raised({
  zDepth = 1,
  radius = 2,
  background = "#fff",
  children,
  styles: passedStyles = {},
}: RaisedProps) {
  const styles = merge<
    Record<string, React.CSSProperties>,
    Record<string, React.CSSProperties>
  >(
    {
      wrap: {
        position: "relative",
        display: "inline-block",
      },
      content: {
        position: "relative",
      },
      bg: {
        position: "absolute",
        inset: "0px",
        boxShadow:
          zDepth === 1
            ? "0 2px 10px rgba(0,0,0,.12), 0 2px 5px rgba(0,0,0,.16)"
            : `0 ${zDepth}px ${zDepth * 4}px rgba(0,0,0,.24)`,
        borderRadius: radius,
        background,
      },
    },
    passedStyles
  );

  return (
    <div style={styles.wrap}>
      <div style={styles.bg} />
      <div style={styles.content}>{children}</div>
    </div>
  );
}
