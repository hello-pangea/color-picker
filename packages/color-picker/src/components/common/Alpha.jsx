import merge from "lodash/merge";
import React, { Component, PureComponent } from "react";
import * as alpha from "../../helpers/alpha";

import Checkboard from "./Checkboard";

export class Alpha extends (PureComponent || Component) {
  componentWillUnmount() {
    this.unbindEventListeners();
  }

  handleChange = (e) => {
    const change = alpha.calculateChange(
      e,
      this.props.hsl,
      this.props.direction,
      this.props.a,
      this.container
    );
    change &&
      typeof this.props.onChange === "function" &&
      this.props.onChange(change, e);
  };

  handleMouseDown = (e) => {
    this.handleChange(e);
    window.addEventListener("mousemove", this.handleChange);
    window.addEventListener("mouseup", this.handleMouseUp);
  };

  handleMouseUp = () => {
    this.unbindEventListeners();
  };

  unbindEventListeners = () => {
    window.removeEventListener("mousemove", this.handleChange);
    window.removeEventListener("mouseup", this.handleMouseUp);
  };

  render() {
    const rgb = this.props.rgb;

    const styles = merge(
      {
        alpha: {
          position: "absolute",
          inset: "0px",
          borderRadius: this.props.radius,
        },
        checkboard: {
          position: "absolute",
          inset: "0px",
          overflow: "hidden",
          borderRadius: this.props.radius,
        },
        gradient: {
          position: "absolute",
          inset: "0px",
          background:
            this.props.direction === "vertical"
              ? `linear-gradient(to bottom, rgba(${rgb.r},${rgb.g},${rgb.b}, 0) 0%,
            rgba(${rgb.r},${rgb.g},${rgb.b}, 1) 100%)`
              : `linear-gradient(to right, rgba(${rgb.r},${rgb.g},${rgb.b}, 0) 0%,
           rgba(${rgb.r},${rgb.g},${rgb.b}, 1) 100%)`,
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: "relative",
          height: "100%",
          margin: "0 3px",
        },
        pointer: {
          position: "absolute",
          left: this.props.direction === "vertical" ? 0 : `${rgb.a * 100}%`,
          top:
            this.props.direction === "vertical" ? `${rgb.a * 100}%` : undefined,
        },
        slider: {
          width: "4px",
          borderRadius: "1px",
          height: "8px",
          boxShadow: "0 0 2px rgba(0, 0, 0, .6)",
          background: "#fff",
          marginTop: "1px",
          transform: "translateX(-2px)",
        },
      },
      this.props.style
    );

    return (
      <div style={styles.alpha}>
        <div style={styles.checkboard}>
          <Checkboard renderers={this.props.renderers} />
        </div>
        <div style={styles.gradient} />
        <div
          style={styles.container}
          ref={(container) => (this.container = container)}
          onMouseDown={this.handleMouseDown}
          onTouchMove={this.handleChange}
          onTouchStart={this.handleChange}
        >
          <div style={styles.pointer}>
            {this.props.pointer ? (
              <this.props.pointer {...this.props} />
            ) : (
              <div style={styles.slider} />
            )}
          </div>
        </div>
      </div>
    );
  }
}

export default Alpha;
