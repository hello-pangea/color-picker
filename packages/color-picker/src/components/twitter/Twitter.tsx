import merge from "lodash/merge";
import React from "react";
import { useColor, withColorProvider } from "../../context/useColor";
import * as color from "../../helpers/color";
import { EditableInput, Swatch } from "../common";

export type TwitterPickerProps = {
  width?: string | number;
  triangle?: "hide" | "top-left" | "top-right";
  colors?: string[];
  styles?: Record<string, React.CSSProperties>;
  className?: string;
};

export const Twitter = ({
  colors = [
    "#FF6900",
    "#FCB900",
    "#7BDCB5",
    "#00D084",
    "#8ED1FC",
    "#0693E3",
    "#ABB8C3",
    "#EB144C",
    "#F78DA7",
    "#9900EF",
  ],
  width = 276,
  triangle = "top-left",
  styles: passedStyles = {},
  className = "",
}: TwitterPickerProps) => {
  const { colors: currentColors, changeColor } = useColor();
  const { hex } = currentColors;

  const styles = merge<
    Record<string, React.CSSProperties>,
    Record<string, React.CSSProperties>
  >(
    {
      card: {
        width,
        background: "#fff",
        border: "0 solid rgba(0,0,0,0.25)",
        boxShadow: "0 1px 4px rgba(0,0,0,0.25)",
        borderRadius: "4px",
        position: "relative",
      },
      body: {
        padding: "15px 9px 9px 15px",
      },
      label: {
        fontSize: "18px",
        color: "#fff",
      },
      triangle: {
        width: "0px",
        height: "0px",
        borderStyle: "solid",
        borderWidth: "0 9px 10px 9px",
        borderColor: "transparent transparent #fff transparent",
        position: "absolute",
        display: triangle === "hide" ? "none" : undefined,
        top:
          triangle === "top-left" || triangle === "top-right"
            ? "-10px"
            : undefined,
        left:
          triangle === "top-left" || triangle === "top-right"
            ? "12px"
            : undefined,
      },
      triangleShadow: {
        width: "0px",
        height: "0px",
        borderStyle: "solid",
        borderWidth: "0 9px 10px 9px",
        borderColor: "transparent transparent rgba(0,0,0,.1) transparent",
        position: "absolute",
        display: triangle === "hide" ? "none" : undefined,
        top:
          triangle === "top-left" || triangle === "top-right"
            ? "-11px"
            : undefined,
        left:
          triangle === "top-left" || triangle === "top-right"
            ? "12px"
            : undefined,
      },
      hash: {
        background: "#F0F0F0",
        height: "30px",
        width: "30px",
        borderRadius: "4px 0 0 4px",
        float: "left",
        color: "#98A1A4",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
      },
      input: {
        width: "100px",
        fontSize: "14px",
        color: "#666",
        border: "0px",
        outline: "none",
        height: "28px",
        boxShadow: "inset 0 0 0 1px #F0F0F0",
        boxSizing: "content-box",
        borderRadius: "0 4px 4px 0",
        float: "left",
        paddingLeft: "8px",
      },
      swatch: {
        width: "30px",
        height: "30px",
        float: "left",
        borderRadius: "4px",
        margin: "0 6px 6px 0",
      },
      clear: {
        clear: "both",
      },
    },
    passedStyles
  );

  const handleChange = (hexcode: string, e: React.MouseEvent) => {
    color.isValidHex(hexcode) &&
      changeColor(
        {
          hex: hexcode,
          source: "hex",
        },
        e
      );
  };

  return (
    <div style={styles.card} className={`twitter-picker ${className}`}>
      <div style={styles.triangleShadow} />
      <div style={styles.triangle} />

      <div style={styles.body}>
        {colors.map((c, i) => {
          return (
            <Swatch
              key={i}
              color={c}
              hex={c}
              style={styles.swatch}
              onClick={handleChange}
              focusStyle={{
                boxShadow: `0 0 4px ${c}`,
              }}
            />
          );
        })}
        <div style={styles.hash}>#</div>
        <EditableInput
          label={null}
          style={{ input: styles.input }}
          value={hex.replace("#", "")}
          onChange={handleChange}
        />
        <div style={styles.clear} />
      </div>
    </div>
  );
};

export default withColorProvider(Twitter);
