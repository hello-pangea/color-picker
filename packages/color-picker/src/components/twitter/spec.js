/* global test, jest, expect */

import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import * as color from "../../helpers/color";

import Twitter from "./Twitter";
import { Swatch } from "../common";

test("Twitter renders correctly", () => {
  const tree = renderer.create(<Twitter {...color.red} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Material renders custom styles correctly", () => {
  const tree = renderer
    .create(
      <Twitter
        {...color.red}
        styles={{ default: { card: { boxShadow: "0 0 10px red" } } }}
      />
    )
    .toJSON();
  expect(tree.props.style.boxShadow).toBe("0 0 10px red");
});

test('Twitter `triangle="hide"`', () => {
  const tree = renderer
    .create(<Twitter {...color.red} triangle="hide" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Twitter `triangle="top-right"`', () => {
  const tree = renderer
    .create(<Twitter {...color.red} triangle="top-right" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Twitter onChange events correctly", () => {
  const changeSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy();
  });
  const tree = mount(<Twitter {...color.red} onChange={changeSpy} />);
  expect(changeSpy).toHaveBeenCalledTimes(0);
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate("click");

  expect(changeSpy).toHaveBeenCalled();
});

test("Twitter with onSwatchHover events correctly", () => {
  const hoverSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy();
  });
  const tree = mount(<Twitter {...color.red} onSwatchHover={hoverSpy} />);
  expect(hoverSpy).toHaveBeenCalledTimes(0);
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate("mouseOver");

  expect(hoverSpy).toHaveBeenCalled();
});
