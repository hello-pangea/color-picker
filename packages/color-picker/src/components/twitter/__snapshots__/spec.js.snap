// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitter \`triangle="hide"\` 1`] = `
<div
  className="twitter-picker "
  style={
    {
      "background": "#fff",
      "border": "0 solid rgba(0,0,0,0.25)",
      "borderRadius": "4px",
      "boxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "position": "relative",
      "width": 276,
    }
  }
>
  <div
    style={
      {
        "borderColor": "transparent transparent rgba(0,0,0,.1) transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "display": "none",
        "height": "0px",
        "left": undefined,
        "position": "absolute",
        "top": undefined,
        "width": "0px",
      }
    }
  />
  <div
    style={
      {
        "borderColor": "transparent transparent #fff transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "display": "none",
        "height": "0px",
        "left": undefined,
        "position": "absolute",
        "top": undefined,
        "width": "0px",
      }
    }
  />
  <div
    style={
      {
        "padding": "15px 9px 9px 15px",
      }
    }
  >
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#FF6900",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#FF6900"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#FCB900",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#FCB900"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#7BDCB5",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#7BDCB5"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#00D084",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#00D084"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#8ED1FC",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#8ED1FC"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#0693E3",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#0693E3"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#ABB8C3",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#ABB8C3"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#EB144C",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#EB144C"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#F78DA7",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#F78DA7"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#9900EF",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#9900EF"
      />
    </span>
    <div
      style={
        {
          "alignItems": "center",
          "background": "#F0F0F0",
          "borderRadius": "4px 0 0 4px",
          "color": "#98A1A4",
          "display": "flex",
          "float": "left",
          "height": "30px",
          "justifyContent": "center",
          "width": "30px",
        }
      }
    >
      #
    </div>
    <div
      style={
        {
          "position": "relative",
        }
      }
    >
      <input
        id="rc-editable-input-3"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        spellCheck="false"
        style={
          {
            "border": "0px",
            "borderRadius": "0 4px 4px 0",
            "boxShadow": "inset 0 0 0 1px #F0F0F0",
            "boxSizing": "content-box",
            "color": "#666",
            "float": "left",
            "fontSize": "14px",
            "height": "28px",
            "outline": "none",
            "paddingLeft": "8px",
            "width": "100px",
          }
        }
        value="22194D"
      />
    </div>
    <div
      style={
        {
          "clear": "both",
        }
      }
    />
  </div>
</div>
`;

exports[`Twitter \`triangle="top-right"\` 1`] = `
<div
  className="twitter-picker "
  style={
    {
      "background": "#fff",
      "border": "0 solid rgba(0,0,0,0.25)",
      "borderRadius": "4px",
      "boxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "position": "relative",
      "width": 276,
    }
  }
>
  <div
    style={
      {
        "borderColor": "transparent transparent rgba(0,0,0,.1) transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "display": undefined,
        "height": "0px",
        "left": "12px",
        "position": "absolute",
        "top": "-11px",
        "width": "0px",
      }
    }
  />
  <div
    style={
      {
        "borderColor": "transparent transparent #fff transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "display": undefined,
        "height": "0px",
        "left": "12px",
        "position": "absolute",
        "top": "-10px",
        "width": "0px",
      }
    }
  />
  <div
    style={
      {
        "padding": "15px 9px 9px 15px",
      }
    }
  >
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#FF6900",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#FF6900"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#FCB900",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#FCB900"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#7BDCB5",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#7BDCB5"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#00D084",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#00D084"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#8ED1FC",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#8ED1FC"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#0693E3",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#0693E3"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#ABB8C3",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#ABB8C3"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#EB144C",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#EB144C"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#F78DA7",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#F78DA7"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#9900EF",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#9900EF"
      />
    </span>
    <div
      style={
        {
          "alignItems": "center",
          "background": "#F0F0F0",
          "borderRadius": "4px 0 0 4px",
          "color": "#98A1A4",
          "display": "flex",
          "float": "left",
          "height": "30px",
          "justifyContent": "center",
          "width": "30px",
        }
      }
    >
      #
    </div>
    <div
      style={
        {
          "position": "relative",
        }
      }
    >
      <input
        id="rc-editable-input-4"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        spellCheck="false"
        style={
          {
            "border": "0px",
            "borderRadius": "0 4px 4px 0",
            "boxShadow": "inset 0 0 0 1px #F0F0F0",
            "boxSizing": "content-box",
            "color": "#666",
            "float": "left",
            "fontSize": "14px",
            "height": "28px",
            "outline": "none",
            "paddingLeft": "8px",
            "width": "100px",
          }
        }
        value="22194D"
      />
    </div>
    <div
      style={
        {
          "clear": "both",
        }
      }
    />
  </div>
</div>
`;

exports[`Twitter renders correctly 1`] = `
<div
  className="twitter-picker "
  style={
    {
      "background": "#fff",
      "border": "0 solid rgba(0,0,0,0.25)",
      "borderRadius": "4px",
      "boxShadow": "0 1px 4px rgba(0,0,0,0.25)",
      "position": "relative",
      "width": 276,
    }
  }
>
  <div
    style={
      {
        "borderColor": "transparent transparent rgba(0,0,0,.1) transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "display": undefined,
        "height": "0px",
        "left": "12px",
        "position": "absolute",
        "top": "-11px",
        "width": "0px",
      }
    }
  />
  <div
    style={
      {
        "borderColor": "transparent transparent #fff transparent",
        "borderStyle": "solid",
        "borderWidth": "0 9px 10px 9px",
        "display": undefined,
        "height": "0px",
        "left": "12px",
        "position": "absolute",
        "top": "-10px",
        "width": "0px",
      }
    }
  />
  <div
    style={
      {
        "padding": "15px 9px 9px 15px",
      }
    }
  >
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#FF6900",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#FF6900"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#FCB900",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#FCB900"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#7BDCB5",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#7BDCB5"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#00D084",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#00D084"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#8ED1FC",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#8ED1FC"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#0693E3",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#0693E3"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#ABB8C3",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#ABB8C3"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#EB144C",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#EB144C"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#F78DA7",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#F78DA7"
      />
    </span>
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          {
            "background": "#9900EF",
            "borderRadius": "4px",
            "cursor": "pointer",
            "float": "left",
            "height": "30px",
            "margin": "0 6px 6px 0",
            "outline": "none",
            "position": "relative",
            "width": "30px",
          }
        }
        tabIndex={0}
        title="#9900EF"
      />
    </span>
    <div
      style={
        {
          "alignItems": "center",
          "background": "#F0F0F0",
          "borderRadius": "4px 0 0 4px",
          "color": "#98A1A4",
          "display": "flex",
          "float": "left",
          "height": "30px",
          "justifyContent": "center",
          "width": "30px",
        }
      }
    >
      #
    </div>
    <div
      style={
        {
          "position": "relative",
        }
      }
    >
      <input
        id="rc-editable-input-1"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        spellCheck="false"
        style={
          {
            "border": "0px",
            "borderRadius": "0 4px 4px 0",
            "boxShadow": "inset 0 0 0 1px #F0F0F0",
            "boxSizing": "content-box",
            "color": "#666",
            "float": "left",
            "fontSize": "14px",
            "height": "28px",
            "outline": "none",
            "paddingLeft": "8px",
            "width": "100px",
          }
        }
        value="22194D"
      />
    </div>
    <div
      style={
        {
          "clear": "both",
        }
      }
    />
  </div>
</div>
`;
